edgelet/edgelet-core/src/authorization.rs

```diff
@@ -16,30 +14,44 @@ pub enum Policy {
     Module(&'static str),
 }
 
-pub struct Authorization<M> {
-    runtime: M,
+#[derive(Clone, Debug, PartialEq)]
+pub enum AuthId {
+    None,
+    Any,
+    Value(String),
+}
+
+impl fmt::Display for AuthId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AuthId::None => write!(f, "none"),
+            AuthId::Any => write!(f, "any"),
+            AuthId::Value(auth_id) => write!(f, "{}", auth_id),
+        }
+    }
+}
+
+pub struct Authorization {
     policy: Policy,
 }
 
-impl<M> Authorization<M>
-where
-    M: 'static + ModuleRuntime,
-    for<'r> &'r <M as ModuleRuntime>::Error: Into<ModuleRuntimeErrorReason>,
-{
-    pub fn new(runtime: M, policy: Policy) -> Self {
-        Authorization { runtime, policy }
+impl Authorization {
+    pub fn new(policy: Policy) -> Self {
+        Authorization { policy }
     }
 
     pub fn authorize(
         &self,
         name: Option<String>,
-        pid: Pid,
+        auth_id: AuthId,
     ) -> impl Future<Item = bool, Error = Error> {
         let name = name.map(|n| n.trim_start_matches('$').to_string());
         match self.policy {
             Policy::Anonymous => Either::A(Either::A(self.auth_anonymous())),
-            Policy::Caller => Either::A(Either::B(self.auth_caller(name, pid))),
-            Policy::Module(ref expected_name) => Either::B(self.auth_module(expected_name, pid)),
+            Policy::Caller => Either::A(Either::B(self.auth_caller(name, auth_id))),
+            Policy::Module(ref expected_name) => {
+                Either::B(self.auth_module(expected_name, auth_id))
```

Comment:
I am not a big fan of these names... From our chat, Module is essentially a whitelist and Caller is are actions a module can perform on itself. These names confuse me because 1)  initially I assumed the exact opposite meaning 2) Everything is a module here so separating module into its own policy name made me think that Caller had nothing to do with modules.

Reply:
I propose:
Policy::Whitelist
Policy::Self

Reply:
I didn't make any changes to the policy. But I agree naming is confusing.